lib/graphqr/query_field.rb

```diff
@@ -45,52 +47,103 @@ def query_field(field_name, active_record_class, type_class:, scope_class: nil,
       is_collection = active_record_class.is_a? Array
       if is_collection
         active_record_class = active_record_class.first
-        resolver = collection(active_record_class, type_class, scope_class)
+        resolver = collection_resolver(active_record_class, type_class, scope_class)
       else
-        resolver = resource(active_record_class, type_class)
+        resolver = resource_resolver(active_record_class, type_class)
       end
 
       field(field_name, paginate: is_collection, resolver: resolver, **kwargs, &block)
     end
-    # rubocop:enable Metrics/ParameterLists
+  end
+
+  module RelationFields
+    include GraphQR
+    ##
+    # DOC PENDING
+    #
+    def has_many(field_name, type_class, scope_class: nil, **kwargs, &block)
+      type_class = type_class.first
+
+      resolver = has_many_resolver(field_name, type_class, scope_class)
+
+      field(field_name, paginate: true, resolver: resolver, **kwargs, &block)
+    end
+
+    def has_one(field_name, type_class, **kwargs, &block)
+      resolver = has_one_resolver(field_name, type_class)
+
+      field(field_name, resolver: resolver, **kwargs, &block)
+    end
+  end
+  # rubocop:enable Metrics/ParameterLists
 
-    private
+  private
```

Comment:
I think you included `GraphQR` in those modules because of this methods..
You could create something like `BaseResourceResolvers` that is define `collection` and `single` resolvers
then in each module you can define your custom resolvers.